src/Kestrel.Core/Internal/Infrastructure/HttpUtilities.cs

```diff
@@ -177,6 +178,87 @@ internal static unsafe HttpMethod GetKnownMethod(byte* data, int length, out int
             return HttpMethod.Custom;
         }
 
+        /// <summary>
+        /// Parses string <paramref name="value"/> for a known HTTP method.
+        /// </summary>
+        /// <remarks>
+        /// A "known HTTP method" can be an HTTP method name defined in the HTTP/1.1 RFC.
+        /// The Known Methods (CONNECT, DELETE, GET, HEAD, PATCH, POST, PUT, OPTIONS, TRACE)
+        /// </remarks>
+        /// <returns><see cref="HttpMethod"/></returns>
+        public static HttpMethod GetKnownMethod(string value)
+        {
+            // Called by http/2 and if user code overrides and sets the Method on IHttpRequestFeature
```

Comment:
This is only used for http2 now, not setting the method directly right?

Reply:
Yes, will change comment